### PR TITLE
PUBDEV-8617: Avoid using sleep() in AutoML job monitoring

### PR DESCRIPTION
**Pull request by michalkurka**
sleep doesn't return the F/J thread to the pool - making it unusable for
other work

### CODE REVIEW

**Comment by michalkurka** [comment]
@sebhrusen something is not right with this PR, I have a failing test `pyunit_automl_early_stopping.py`

- it is finishing in the given timeout but it builds more models than allowed?!

File: h2o-core/src/main/java/water/Job.java [hunk 1]
```diff
@@ -510,15 +510,21 @@ protected void copyOver(Job src) {
...
     return j;
   }
 
-  void waitForDone(long timeMillis) {
+  void waitForDone(long timeMillis, boolean checkConsistency) {
     final Barrier2 bar = _barrier;
-    if (bar == null)
+    if (bar == null) {
+      if (checkConsistency) {
+        if (isRunning()) { // barrier is only removed after job is stopped
```
Comment (sebhrusen) [inline h2o-core/src/main/java/water/Job.java:522]: you like nested ifs, don't you? :)

**Review by sebhrusen — APPROVED** [review]
Thanks @michalkurka 
That's a nice improvement for AutoML on small datasets, especially as we see some competitors (flaml if I'm correct) now starting to promote training speed.

**Review by sebhrusen — APPROVED** [review]
Looks great! Thanks a lot for diving into this... was not as simple as expected I guess.

File: scripts/jenkins/Makefile.jenkins [hunk 2]
```diff
@@ -105,7 +105,13 @@ test-pyunit-xgboost-stress:
...
+	bash -c 'echo "XGB:ON,POLL:SLEEP"; for i in `seq 5`; do (cd h2o-py/tests/ && ../../scripts/run.py --wipeall --numclouds 1 --jvm.xmx 4g --test pyunit_automl_input.py --jvm.opts "-Dsys.ai.h2o.ext.core.toggle.XGBoost=true -Dsys.ai.h2o.job.sleep_wait_for_done=true") | grep "pass.*pyunit"; done'
+	bash -c 'echo "XGB:ON,POLL:WAIT"; for i in `seq 5`; do (cd h2o-py/tests/ && ../../scripts/run.py --wipeall --numclouds 1 --jvm.xmx 4g --test pyunit_automl_input.py --jvm.opts "-Dsys.ai.h2o.ext.core.toggle.XGBoost=true -Dsys.ai.h2o.job.sleep_wait_for_done=false") | grep "pass.*pyunit"; done'
+	bash -c 'echo "XGB:OFF,POLL:SLEEP"; for i in `seq 5`; do (cd h2o-py/tests/ && ../../scripts/run.py --wipeall --numclouds 1 --jvm.xmx 4g --test pyunit_automl_input.py --jvm.opts "-Dsys.ai.h2o.ext.core.toggle.XGBoost=false -Dsys.ai.h2o.job.sleep_wait_for_done=true") | grep "pass.*pyunit"; done'
+	bash -c 'echo "XGB:OFF,POLL:WAIT"; for i in `seq 5`; do (cd h2o-py/tests/ && ../../scripts/run.py --wipeall --numclouds 1 --jvm.xmx 4g --test pyunit_automl_input.py --jvm.opts "-Dsys.ai.h2o.ext.core.toggle.XGBoost=false -Dsys.ai.h2o.job.sleep_wait_for_done=false") | grep "pass.*pyunit"; done'
```
[the remark this answers is not in the document]
Comment (michalkurka) [reply scripts/jenkins/Makefile.jenkins]: thank you, removed

**Pull request merged**